iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java

```diff
@@ -111,6 +114,26 @@ public TargetScopeDeclaration TARGET_SCOPE_DECLARATION() {
         EXPRESSION()));
   }
 
+  // TODO SONARIAC-962 Put in place decorator
+  public ParameterDeclaration PARAMETER_DECLARATION() {
+    return b.<ParameterDeclaration>nonterminal(BicepLexicalGrammar.PARAMETER_DECLARATION).is(
+      b.firstOf(
+        f.parameterDeclaration(
+          b.token(BicepKeyword.PARAMETER),
+          IDENTIFIER(),
+          b.token(BicepKeyword.RESOURCE),
+          INTERPOLATED_STRING_TYPE(),
+          b.optional(b.token(Punctuator.EQU)),
+          b.optional(EXPRESSION())),
```

Comment:
1. The `EQU` and `EXPRESSION()` as a whole need to be optional. Currently, the following input will be accepted:
`parameter id resource iterString =` (with equals but without expression at the end).
2. Extraction to separate method will help - just to be as much as possible with original grammar, and reuse `parameterDefaultValue` also in lines 133-134

Reply:
Not necessary, I got the same use case with Docker and Nils suggested that we don't need to make them whole (see `Flag` from Docker for example).
We tend to enforce the grammar/syntax, but we are not writing a validator :smile:
When writing our parser we suppose that the provided code is valid, so we don't need to ensure/enforce this. There would be no use case where a valid code would be parsed into an invalid AST or trigger a parse error here.

I wanted to avoid introducing `parameterDefaultValue` grammar element, from my opinion it was just adding a layer of complexity (+ method in grammar + classes to handle it + tests) for not much added value.

WDYT?